src/modules/Game.class.js

```diff
@@ -20,25 +20,249 @@ class Game {
    * If passed, the board will be initialized with the provided
    * initial state.
    */
-  constructor(initialState) {
-    // eslint-disable-next-line no-console
-    console.log(initialState);
+  constructor(
+    initialState = [
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+      [0, 0, 0, 0],
+    ],
+  ) {
+    this.board = initialState;
+    this.firstMove = false;
+    this.previousBoard = JSON.parse(JSON.stringify(this.board));
+    this.initControls();
+    this.getState();
   }
 
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
+  hasBoardChanged() {
+    return JSON.stringify(this.board) !== JSON.stringify(this.previousBoard);
+  }
+
+  initControls() {
+    document.addEventListener('keydown', (e) => {
+      if (e.key === 'ArrowLeft') {
+        this.moveLeft();
+      } else if (e.key === 'ArrowRight') {
+        this.moveRight();
+      } else if (e.key === 'ArrowUp') {
+        this.moveUp();
+      } else if (e.key === 'ArrowDown') {
+        this.moveDown();
+      }
+    });
+  }
+
+  moveLeft() {
+    for (let row = 0; row < 4; row++) {
+      let newRow = this.board[row].filter((num) => num !== 0);
+
+      for (let i = 0; i < newRow.length - 1; i++) {
+        if (newRow[i] === newRow[i + 1]) {
+          newRow[i] *= 2;
+          newRow[i + 1] = 0;
+        }
+      }
+
+      newRow = newRow.filter((num) => num !== 0);
+
+      while (newRow.length < 4) {
+        newRow.push(0);
+      }
+      this.board[row] = newRow;
+    }
+
+    if (this.hasBoardChanged()) {
+      this.previousBoard = JSON.parse(JSON.stringify(this.board));
+      this.addNewTile();
+      this.getState();
+      this.checkFirstMove();
+      this.getScore();
+
+      const gameStatus = this.getStatus();
+      const messageWin = document.querySelector('.message-win');
+      const messageLose = document.querySelector('.message-lose');
+
+      if (gameStatus === 'win') {
+        messageWin.classList.remove('hidden');
```

Comment:
This logic is similar in every method, you need to save it in a helper function and use it everywhere.